tests/test_code_report.py

```diff
@@ -10,49 +10,64 @@
+class ConfigData:
+    def __init__(self):
+        self.text = "from universum.configuration_support import Configuration, Step\n"
+        self.text += "configs = Configuration()\n"
 
-        configs = Configuration([dict(name="Run static pylint", code_report=True,
-            command=['{python()}', '-m', 'universum.analyzers.pylint'{''.join(args)}])])
-    """)
+    def add_analyzer(self, analyzer: str, arguments: List[str]) -> 'ConfigData':
+        args = [f", '{arg}'" for arg in arguments]
+        cmd = f"['{python()}', '-m', 'universum.analyzers.{analyzer}'{''.join(args)}]"
+        self.text += f"configs += Configuration([Step(name='Run {analyzer}', code_report=True, command={cmd})])\n"
+        return self
+
+    def finalize(self) -> str:
+        return inspect.cleandoc(self.text)
 
 
 source_code = """
 "Docstring."
 
-print("Hello world.")
+s: str = "world"
+print(f"Hello {s}.")
 """
 
 log_fail = r'Found [0-9]+ issues'
-log_success = r'Issues not found.'
-
-
-@pytest.mark.parametrize('args, tested_content, expected_log', [
-    [["--result-file", "${CODE_REPORT_FILE}"], source_code, log_success],
-    [["--result-file", "${CODE_REPORT_FILE}"], source_code + '\n', log_fail]
+log_success = r'Issues not found'
 
+@pytest.mark.parametrize('analyzers, common_args, tested_content, expected_success', [
+    [['pylint', 'mypy'], ["--result-file", "${CODE_REPORT_FILE}"], source_code, True],
+    [['pylint'], ["--result-file", "${CODE_REPORT_FILE}"], source_code + '\n', False],
+    [['mypy'], ["--result-file", "${CODE_REPORT_FILE}"], source_code.replace(': str', ': int'), False],
+    [['pylint', 'mypy'], ["--result-file", "${CODE_REPORT_FILE}"], source_code.replace(': str', ': int') + '\n', False],
     # TODO: add test with rcfile
     # TODO: parametrize test for different versions of python
 ])
-def test_code_report(runner_with_pylint, args, tested_content, expected_log):
-    runner_with_pylint.local.root_directory.join("source_file.py").write(tested_content)
-    config = get_config(["--python-version", python_version(), "--files", "source_file.py"] + args)
-
-    log = runner_with_pylint.run(config)
-    assert re.findall(expected_log, log)
-
-
-def test_without_code_report_command(runner_with_pylint):
-    log = runner_with_pylint.run("""
+def test_code_report_log(runner_with_analyzers, analyzers, common_args, tested_content, expected_success):
+    common_args.extend(["--python-version", python_version(), "--files", "source_file.py"])
+    runner_with_analyzers.local.root_directory.join("source_file.py").write(tested_content)
+    config = ConfigData()
+    for analyzer in analyzers:
+        config.add_analyzer(analyzer, common_args)
+
+    log = runner_with_analyzers.run(config.finalize())
+    if expected_success:
+        assert re.findall(log_success, log)
+    else:
+        assert re.findall(log_fail, log)
+        for analyzer in analyzers:
+            assert re.findall(fr'Run {analyzer} - [^\n]*Failed', log)
```

Comment:
to make sure that second analyzer is checked even if first failed

Reply:
Maybe add this as a comment to source code?

Reply:
done